### DISCUSSION TITLE
How to get the error code of error?

### DISCUSSION THREAD

**Discussion by yeungc** [wevm/wagmi#233] [General]
Hello guys, I would like to get the error code of error from useContractWrite, however I can't access it from the error object, please check below snippet:

```
  const [{ data: contractWriteData, error: contractWriteError }, write] = useContractWrite(
    {
      addressOrName: CONTRACT_ADDRESS,
      contractInterface: CONTRACT_ABI,
    },
    'pay'
  )

  useEffect(() => {
    contractWriteError && console.log(contractWriteError) // To print the contractWriteError object
  }, [contractWriteError])

  useEffect(() => {
    contractWriteError && contractWriteError.code // This one got an error alert (Property 'code' does not exist on type 'Error')
  }, [contractWriteError])
```

Based on result of console.log(contractWriteError), I can see that the contractWriteError object has error code actually:
![0](https://user-images.githubusercontent.com/7931068/156897689-5ea22dd3-e6a6-4632-a3e1-2eb862bb7788.png)

However I can't access it:
![1](https://user-images.githubusercontent.com/7931068/156897835-5ca75e7a-2225-453f-a895-8212d2661c8c.png)

May I ask what should I do to get the error code? Thanks!

**Comment by tmm** [comment]
Unfortunately, ethers doesn't provide a type for their errors and just extends `Error`:

https://github.com/ethers-io/ethers.js/blob/master/packages/logger/lib/index.js#L232-L235=
```ts
// @TODO: Any??
var error = new Error(message);
error.reason = reason;
error.code = code;
```

**Reply by a-znk** [reply]
@tmm would it be possible to use `reason`

```
{(isPrepareError || isMintError) && (
<div className={mintStyles.mint_error}>
Error: {(prepareError || mintError)?.reason}
</div>
)}
```

instead of `message?`

```
{(isPrepareError || isMintError) && (
<div className={mintStyles.mint_error}>
Error: {(prepareError || mintError)?.message}
</div>
)}
```

Using `reason` instead of `message` displays a much easier to understand error for the end user. However, I do receive the following message in VSCode. Any thoughts?

> "Property 'reason' does not exist on type 'Error'

**Comment by terrytjw** [comment]
Type your `error` as `any` for now I guess.

**Reply by tmm** [reply]
You can also import `RpcError` from `wagmi`/`@wagmi/core` and use that instead of `any`.

**Reply by terrytjw** [reply]
Don't seem to be working for me
<img width="486" alt="image" src="https://user-images.githubusercontent.com/59902670/209360409-d9dcd8a3-a2d8-467b-8a23-e1ff315b9cd2.png">

**Reply by tmm** [reply]
I meant that you can either use the `wagmi` package or `@wagmi/core`.

**Reply by terrytjw** [reply]
What about the `onError` parameter in `usePrepareContractWrite()`? I've been using `any` so that I can actually access the error properties for some error handling (prompting users of insufficient funds and switching to the right network). How would you go about this? Or is there a better way for me to do error handling? 
<img width="698" alt="image" src="https://user-images.githubusercontent.com/59902670/209361309-69a3870f-9216-4f28-ab18-c45aec9636c2.png">

**Reply by Nilbristi** [reply]
> You can also import `RpcError` from `wagmi`/`@wagmi/core` and use that instead of `any`.
@tmm How this function will look in Angular? Can you give me a code example?
